config/config.go

```diff
@@ -306,11 +306,9 @@ func parseCreationRuleForFile(conf *configFile, filePath string, kmsEncryptionCo
 			rule = &r
 			break
 		}
-		if r.PathRegex != "" {
-			if match, _ := regexp.MatchString(r.PathRegex, filePath); match {
-				rule = &r
-				break
-			}
+		if regexp.MustCompile(r.PathRegex).MatchString(filePath) {
```

Comment:
The `if r.PathRegex != "" {` is missing.

Also, I'm not sure whether using `MustCompile` here is a good idea in terms of error handling. It will panic if `r.PathRegex` does not compile. The old code was simply ignoring errors, which might not be the be solution, but panicing instead is not a good idea IMO.

Reply:
The `r.PathRegex != ""` is not required because the opposite case is already tested before, right? ^^
I'll modify the code to handle the compilation error.

Reply:
> The `if r.PathRegex != "" {` is missing.
> 
> Also, I'm not sure whether using `MustCompile` here is a good idea in terms of error handling. It will panic if `r.PathRegex` does not compile. The old code was simply ignoring errors, which might not be the be solution, but panicing instead is not a good idea IMO.

The `if r.PathRegex != "" {` is useless because we check if the string is empty before, and if it's the case, then  it breaks. So the string is not empty at this point.
I update the code so I return an error instead of panicing.